task-plugins/dioptra_builtins/registry/art.py

```diff
@@ -59,28 +61,45 @@
 @require_package("art", exc_type=ARTDependencyError)
 @require_package("tensorflow", exc_type=TensorflowDependencyError)
 def load_wrapped_tensorflow_keras_classifier(
-    name: str, version: int, classifier_kwargs: Optional[Dict[str, Any]] = None
-) -> KerasClassifier:
+    name: str,
+    version: int,
+    imagenet_preprocessing: bool = False,
+    classifier_kwargs: Optional[Dict[str, Any]] = None,
+) -> TensorFlowV2Classifier:
     """Loads and wraps a registered Keras classifier for compatibility with the |ART|.
 
     Args:
         name: The name of the registered model in the MLFlow model registry.
         version: The version number of the registered model in the MLFlow registry.
         classifier_kwargs: A dictionary mapping argument names to values which will
-            be passed to the KerasClassifier constructor.
+            be passed to the TensorFlowV2Classifier constructor.
     Returns:
-        A trained :py:class:`~art.estimators.classification.KerasClassifier` object.
+        A trained :py:class:`~art.estimators.classification.TensorFlowV2Classifier`
+            object.
 
     See Also:
-        - :py:class:`art.estimators.classification.KerasClassifier`
+        - :py:class:`art.estimators.classification.TensorFlowV2Classifier`
         - :py:func:`.mlflow.load_tensorflow_keras_classifier`
     """
     classifier_kwargs = classifier_kwargs or {}
     keras_classifier: Sequential = load_tensorflow_keras_classifier(
         name=name, version=version
     )
-    wrapped_keras_classifier: KerasClassifier = KerasClassifier(
-        model=keras_classifier, **classifier_kwargs
+    nb_classes = keras_classifier.output_shape[1]
+    input_shape = keras_classifier.input_shape
+    loss_object = losses.get(keras_classifier.loss)
+    preprocessing = (
+        (np.array([103.939, 116.779, 123.680]), np.array([1.0, 1.0, 1.0]))
+        if imagenet_preprocessing
+        else None
+    )
```

Comment:
I agree it's not ideal.

This is actually one of the things I consolidated from another spot. There were a few other `load_tensorflow_keras_classifier` functions scattered in the custom plugins and I opted to update those to use the builtins (this was already done for fgsm).

One example here:
https://github.com/usnistgov/dioptra/blob/d39c986c9933b87de6f1eda7668ed65d23da2fc8/examples/task-plugins/dioptra_custom/custom_patch_plugins/registry_art.py#L55

Reply:
Good to know, that's good that this is at least consolidating the logic into a single place so that when we come up with a better solution we only need to change it in one place.